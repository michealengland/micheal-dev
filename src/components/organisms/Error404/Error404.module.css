.error404 {
  @apply bg-primary-purple1 text-nuetral-white min-h-screen flex justify-center items-center flex-col;

  & .title {
    @apply text-3xl md:text-7xl mb-12 text-tertiary-teal max-w-3xl text-center;
  }

  & .content {
    @apply text-xl mb-12;
  }
}
