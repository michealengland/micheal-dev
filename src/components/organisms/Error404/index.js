import React from 'react'
import {Link} from 'gatsby'
import PropTypes from 'prop-types'

export default function Error404({content, title}) {
  return (
    <section>
      <title>{title}</title>
      <h1>{title}</h1>
      <p>{content}</p>
      <Link to="/">Go home</Link>
    </section>
  )
}

Error404.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string
}
