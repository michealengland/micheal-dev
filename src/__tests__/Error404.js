import React from 'react'
import renderer from 'react-test-renderer'
import Error404 from '@components/organisms/Error404'
describe('Error404', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Error404
          title="404 page not found"
          content="This page is missing or does not exist."
          img="./images/adrien-converse-REL986jyuX4-unsplash.jpg"
          label="Back to Micheal.dev"
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
