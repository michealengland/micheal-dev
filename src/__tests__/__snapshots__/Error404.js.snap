// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 renders correctly 1`] = `
<section
  className="error404"
  style={
    Object {
      "backgroundImage": "url(./images/adrien-converse-REL986jyuX4-unsplash.jpg)",
    }
  }
>
  <title>
    404 page not found
  </title>
  <h1
    className="title"
  >
    404 page not found
  </h1>
  <p
    className="content"
  >
    This page is missing or does not exist.
  </p>
  <a
    className="button"
    href="/"
  >
    Back to Micheal.dev
  </a>
</section>
`;
